server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+            .matcher(All.class)
+            .mailet(DKIMSign.class)
+            .addProperty(
+                    "signatureTemplate",
+                    "v=1; s=selector; d=example.com; h=from:to:received:received; a=rsa-sha256; bh=; b=;")
+            .addProperty("privateKey", TESTING_PEM)
+            .build();
+    private static final MailetConfiguration DKIMVERIFY_MAILET = MailetConfiguration.builder()
+            .matcher(All.class)
+            .mailet(DKIMVerify.class)
+            .build();
+    private static final MailetConfiguration STUB_MAILET = MailetConfiguration.builder()
+            .matcher(All.class)
+            .mailet(ExtractAttributeStub.class)
+            .addProperty("attributeName", DKIMVerify.DKIM_AUTH_RESULT.asString())
+            .build();
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Rule
+    public IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+    @Rule
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+
+    private TemporaryJamesServer jamesServer;
+    private DataProbe dataProbe;
+    private List<Optional<String>> dkimAuthResults;
+
+    @Before
+    public void setup() throws Exception {
+        dkimAuthResults = new ArrayList<>();
+        ExtractAttributeStub.setDkimAuthResultInspector(value -> dkimAuthResults.add(value.map(result -> (String) result)));
+        MailetContainer.Builder mailetContainer = TemporaryJamesServer.SIMPLE_MAILET_CONTAINER_CONFIGURATION
+                .putProcessor(ProcessorConfiguration.transport()
+                        .addMailet(DKIMSIGN_MAILET)
+                        .addMailet(DKIMVERIFY_MAILET)
+                        .addMailet(STUB_MAILET)
+                        .addMailetsFrom(CommonProcessors.transport()));
+
+        jamesServer = TemporaryJamesServer
+                .builder()
+                .withMailetContainer(mailetContainer)
+                .build(temporaryFolder);
+
+        dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+
+        dataProbe.addUser(RECIPIENT, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void incomingMessageShouldBeReceivedSigned() throws Exception {
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .sendMessage(FROM, RECIPIENT);
+
+        imapMessageReader.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT, PASSWORD)
+            .select(IMAPMessageReader.INBOX)
+            .awaitMessage(awaitAtMostOneMinute);
+
+        assertThat(dkimAuthResults).hasSize(1);
+        assertThat(dkimAuthResults.get(0)).hasValueSatisfying(result -> assertThat(result).startsWith("pass"));
+        assertThat(imapMessageReader.readFirstMessage()).isNotNull();
```

Comment:
why don't we build assertions on the mail we find with IMAP?

Reply:
That's a WIP, I was supposed to get the right value

Reply:
My question was: why use a ExtractAttributeStub at all when you can read the headers in the mail you retrieve with IMAP?